src/Symfony/Bridge/PhpUnit/ErrorAssert.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Symfony\Bridge\PhpUnit;
+
+/**
+ * Test that your code triggers expected error messages.
+ *
+ * @author Christian Flothmann <christian.flothmann@xabbuh.de>
+ */
+final class ErrorAssert
+{
+    public static function assertDeprecationsAreTriggered($expectedMessages, callable $testCode)
+    {
+        self::assertErrorsAreTriggered(E_USER_DEPRECATED, $expectedMessages, $testCode);
+    }
+
+    public static function assertErrorsAreTriggered($expectedType, $expectedMessages, callable $testCode)
+    {
+        $triggeredMessages = array();
+
+        set_error_handler(function ($type, $message) use ($expectedType, &$triggeredMessages) {
+            if ($expectedType !== $type) {
+                restore_error_handler();
```

Comment:
Because an exception is thrown and the function will stop here.
In case there are other tests the default handler must be restored.